### PR TITLE
feat/AB#82897_new_grid_type_on_dashboard

### PR DESCRIPTION
**Pull request by RenzoPrats**
# Description

Feat: Enabled to use "fit" grid type in dashboard.

## Useful links

https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/82897

## Type of change

- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

As the attached video.

## Screenshots

Peek 08-01-2024 18-23
Peek 09-01-2024 16-53

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by AntoineRelief** [comment]
@RenzoPrats 
there is one issue I'll try to fix an issue with the tabs widget
I think the size of the grid is not calculated well. See screenshot:
Capture d’écran 2024-01-09 à 13 28 20

The widgets are outside of the container, but they should fit in it
also, there is vertical scroll, but the widgets should also fit vertically in the container

File: libs/shared/src/lib/components/widget-grid/widget-grid.component.ts [hunk 1]
```diff
@@ -165,6 +175,19 @@ export class WidgetGridComponent
     }
   }
 
+  /**
+   * Set the available width and height for fit grid type
+   */
+  public setFitStyleWidth() {
+    // all available space less left sidenav
+    this.fitStyle['max-width.px'] = document.documentElement.clientWidth - 300;
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/widget-grid/widget-grid.component.ts]: @RenzoPrats 
I think the calculation there is wrong
you are doing it based on document, but check again the description of the ticket, this is not what I was expecting

File: libs/shared/src/lib/components/widget-grid/widget-grid.component.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<ng-container *ngIf="!loading; else emptyTemplate">
+<ng-container *ngIf="!loading; else emptyTemplate" #teste>
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/widget-grid/widget-grid.component.html]: @RenzoPrats you got an artifact from your test there I believe
this should be removed

**Review by Maximo-reliefapps — COMMENTED** [review]
Looks good to me
There is an issue with the ui-select-menu component, which doesn't show the selected option translated at first, but it's unrelated to the ticket

**Comment by AntoineRelief** [comment]
@Maximo-reliefapps 
please have another look
from the videos shared by Renzo I can already see there is an issue, for the tabs widget:

Capture d’écran 2024-01-10 à 20 20 25

you can see the width behind the modal for the grid is not correct

**Comment by AntoineRelief** [comment]
@Maximo-reliefapps I'm putting back as draft the PR,
when ready, please remove draft status, thanks!

**Comment by Maximo-reliefapps** [comment]
Here is a video with all cases fixed.
There were size issues in full screen dashboards, full screen tab widgets and tab widget settings.

https://github.com/ReliefApplications/ems-frontend/assets/94831019/e894650b-a643-4d21-994b-b0f538645230

Variable filter size continues to work. Everything is working on the front office too.
There is maybe some stuff to modify for the web widgets, I haven't tested them but if they have the capability to expand the dashboard, we need to pass the fullscreen property.

**Comment by AntoineRelief** [comment]
@Maximo-reliefapps 
I doesn't look very well
it seems a bit "lazy" to expand no? 😅👀

**Comment by Maximo-reliefapps** [comment]
@AntoineRelief 
It was the least intrusive solution, it would be better to have the height set up automatically based on the remaining space, but in some of the places we don't have fixed heights, as in the dashboard.
I will work on it again and see if I can find a better solution.

**Pull request closed without merging**